flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/optimize/program/FlinkRecomputeStatisticsProgram.java

```diff
@@ -115,25 +115,26 @@ private TableStats recomputeStatistics(
             FilterPushDownSpec filterPushDownSpec,
             boolean reportStatEnabled) {
         TableStats origTableStats = table.getStatistic().getTableStats();
+        // bail out early if stats computation is disabled.
+        if (!reportStatEnabled) {
+            return origTableStats;
+        }
         DynamicTableSource tableSource = table.tableSource();
         if (filterPushDownSpec != null && !filterPushDownSpec.isAllPredicatesRetained()) {
             // filter push down but some predicates are accepted by source and not in reaming
             // predicates
             // the catalog do not support get statistics with filters,
-            // so only call reportStatistics method if reportStatEnabled is true
+            // so only call reportStatistics
```

Comment:
nit: the comment now does not read well.  "so only call reportStatistics" implies there will be more to the sentence.